src/functions.rs

```diff
@@ -1770,6 +1770,68 @@ impl fmt::Display for FunctionArgument<'_> {
     }
 }
 
+fn validate_cast(
+    s: &CascadeString,
+    cast_ti: Option<&TypeInfo>,
+    func_call: Option<&FuncCall>,
+    types: &TypeMap,
+    context: &BlockContext,
+    file: Option<&SimpleFile<String, String>>,
+) -> Result<(), ErrorItem> {
+    let type_info = types.get(s.as_ref());
+    if type_info.is_none()
+        && !context
+            .symbol_in_context(s.as_ref(), types)
+            .map(|ti| ti.is_setype(types))
+            .unwrap_or(false)
+    {
+        return Err(ErrorItem::make_compile_or_internal_error(
+            "Cannot typecast",
+            file,
+            s.get_range(),
+            "This is not something that can be typecast",
+        ));
+    } else if let Some(cast_type_info) = cast_ti {
```

Comment:
cast_ti is also None for the validate_argcast stuff, since it was not originally used.  So right now, being None is okay.

Reply:
I think we should pass cast_ti through validate_argcast from validate_argument.  It looks like validate_argument checks the validity of the cast target directly.  That logic should probably move into validate_cast and apply to both situations?

Reply:
It looks like all we're using the TypeInstance for is to check if it matches TypeInstance::SEType.  That should be equivalent to calling is_setype() on the TypeInfo, I think?

Ideally, casting should work without any need to rely on TypeInstances, since we care what these things *are* not what they're *called*.

Reply:
Looking at the latest version of the code, I think the question about cast_ti being None still stands.  The validate_argcast case has become guaranteed to be Some(), and the ValidatedCall case is the return of types.get().

Without even thinking about callers though, I'm not sure how doing any casting when the target is None makes sense, so it should probably be an error.

Reply:
The None case for cast_ti should be handled at line 1824 in functions.rs.